platform/linux-generic/odp_schedule_basic.c

```diff
@@ -1281,40 +1281,40 @@ static inline int schedule_grp_prio(odp_queue_t *out_queue, odp_event_t out_ev[]
 		num = _odp_sched_queue_deq(qi, ev_tbl, max_deq, !pktin);
 
 		if (odp_unlikely(num < 0)) {
-			/* Destroyed queue. Continue scheduling the same
-			 * priority queue. */
+			/* Remove destroyed queue from scheduling. Continue scheduling
+			 * the same group/prio/spread. */
+			continue;
+		}
+
+		if (num == 0 && !pktin) {
```

Comment:
This is not a strong opinion, but I think having the check for !pktin inside the following if statement was more readable. Now I have to keep more state in my head when reading the code.

Reply:
Agree, !pktin in side if (num == 0) is clear. I changed to that. It decreases 6% performance in one of my test cases in one server system, but increases a bit on another.

```if (num == 0) {
	int direct_recv = !ordered;
	int num_pkt;

	if (!pktin) {
		/* Remove empty queue from scheduling */
		continue;
	}

	/* Poll packet input queue */
	num_pkt = poll_pktin(qi, direct_recv, ev_tbl, max_deq);```